src/control/types.ts

```diff
@@ -30,3 +30,21 @@ export const MetadataConfigSchema = Type.Object(
 // string. To avoid this confusing case, we require lenth > 1.
 export const CollectionNameSchema = nonemptyString;
 export type CollectionName = string;
+
+/**
+ * @see [Understanding indexes](https://docs.pinecone.io/docs/indexes)
+ */
+export type PodType =
+  | 's1.x1'
+  | 's1.x2'
+  | 's1.x4'
+  | 's1.x8'
+  | 'p1.x1'
+  | 'p1.x2'
+  | 'p1.x4'
+  | 'p1.x8'
+  | 'p2.x1'
+  | 'p2.x2'
+  | 'p2.x4'
+  | 'p2.x8'
+  | string;
```

Comment:
This type includes `string` so isn't really a narrowing of the type in comparison to the past. But I thought it would make for better documentation to show that actual valid values here.